src/slashCommand/slashCommand.service.ts

```diff
@@ -0,0 +1,68 @@
+import { Injectable, OnModuleInit } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import {
+  REST,
+  RESTPostAPIApplicationCommandsJSONBody,
+  Routes,
+} from 'discord.js';
+import * as fs from 'fs';
+import { BaseCommand } from './slashCommand.module';
+
+@Injectable()
+export class SlashCommandService implements OnModuleInit {
+  private rest: REST;
```

Comment:
I think it would be nice to have this in the discord service and then expose a method to register the slash commands.

Reply:
I made an attempt to implement this suggestion - maybe have a look if it makes sense to you?